src/blueprint/blueprint.rs

```diff
@@ -219,4 +221,61 @@ impl Blueprint {
     // generation of schema cannot fail.
     schema.finish().unwrap()
   }
+
+  /// Generate a dumb schema useful for validation.
+  pub fn to_validation_schema(&self) -> Result<Schema, ValidationError<String>> {
```

Comment:
let's move all the validation stuff into separate file `blueprint/validation.rs`